tensorflow/core/lib/gtl/compactptrset.h

```diff
@@ -19,6 +19,12 @@ limitations under the License.
 #include <type_traits>
 #include "tensorflow/core/lib/gtl/flatset.h"
 
+#ifdef _WIN64
```

Comment:
There's a variant of this defined in a test file here:
https://github.com/tensorflow/tensorflow/blob/5061cc82f21a8fbb71751affec104382e38b777f/tensorflow/core/lib/strings/strcat_test.cc#L25

I think I like the other definition that uses `ptrdiff_t` better.  Let's put the common definition under here, or a windows-specific integral_types.h file.
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/platform/default/integral_types.h

Reply:
To be clear, I'm thinking we should add a file `tensorflow/core/platform/windows/integral_types.h` that includes `tensorflow/core/platform/default/integral_types.h`, and adds `typedef ptrdiff_t ssize_t`.

And we change `tensorflow/core/platform/types.h` to include `windows/integral_types.h` instead of `default/integral_types.h` https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/platform/types.h

Let's see what @mrry  thinks of this, or if there are any issues with this suggestion.

Reply:
+1 to @tatatodd's suggestion.

Reply:
Done.